allennlp/tests/data/dataset_readers/universal_dependencies_multilingual_dataset_reader_test.py

```diff
@@ -0,0 +1,112 @@
+            assert fields1["head_indices"].labels == [0, 1, 5, 5, 2, 9, 6, 6, 1, 12, 12, 9, 1]
+
+            assert [t.text for t in fields2["words"].tokens] == ['Es', 'un', 'bar', 'disfrazado', 'de',
+                                                                 'restaurante', 'la', 'comida', 'esta',
+                                                                 'demasiado', 'salada', '.']
+            assert fields2["pos_tags"].labels == ['VERB', 'DET', 'NOUN', 'VERB', 'ADP', 'NOUN',
+                                                  'DET', 'NOUN', 'VERB', 'PRON', 'ADJ', '.']
+            assert fields2["head_tags"].labels == ['ROOT', 'det', 'attr', 'partmod', 'adpmod', 'adpobj',
+                                                   'det', 'nsubj', 'parataxis', 'nmod', 'acomp', 'p']
+            assert fields2["head_indices"].labels == [0, 3, 1, 3, 4, 5, 8, 9, 1, 11, 9, 1]
+
+        elif lang == 'it':
+            assert fields1['metadata'].metadata['lang'] == 'it'
+            assert [t.text for t in fields1["words"].tokens] == ['Inconsueto', 'allarme', 'alla', 'Tate',
+                                                                 'Gallery', ':']
+            assert fields1["pos_tags"].labels == ['ADJ', 'NOUN', 'ADP', 'NOUN', 'NOUN', '.']
+            assert fields1["head_tags"].labels == ['amod', 'ROOT', 'adpmod', 'dep', 'adpobj', 'p']
+            assert fields1["head_indices"].labels == [2, 0, 2, 5, 3, 2]
+
+            assert fields2['metadata'].metadata['lang'] == 'it'
+            assert [t.text for t in fields2["words"].tokens] == ['Hamad', 'Butt', 'è', 'morto', 'nel', '1994',
+                                                                 'a', '32', 'anni', '.']
+            assert fields2["pos_tags"].labels == ['NOUN', 'NOUN', 'VERB', 'VERB', 'ADP',
+                                                  'NUM', 'ADP', 'NUM', 'NOUN', '.']
+            assert fields2["head_tags"].labels == ['dep', 'nsubj', 'aux', 'ROOT', 'adpmod', 'adpobj',
+                                                   'adpmod', 'num', 'adpobj', 'p']
+            assert fields2["head_indices"].labels == [2, 4, 4, 0, 4, 5, 4, 9, 7, 4]
+
+        return lang
+
+    def test_read_from_files_first_pass_true(self):
+        reader = UniversalDependenciesMultiLangDatasetReader(
+                languages=['es', 'fr', 'it'], is_first_pass_for_vocab=True)
+        instances = list(reader.read(str(self.data_path)))
+        assert len(instances) == 6
+
+        processed_langs = []
+
+        processed_langs.append(self.check_two_instances(instances[0], instances[1]))
+        processed_langs.append(self.check_two_instances(instances[2], instances[3]))
+        processed_langs.append(self.check_two_instances(instances[4], instances[5]))
+
+        assert 'es' in processed_langs and 'fr' in processed_langs and 'it' in processed_langs
+
+    def test_read_from_files_first_pass_false(self):
+        '''
+        Note: assumes that each data file contains two trees
```

Comment:
You might consider increasing the count threshold to 20 or something, just to make the test more robust to any future changes.  It also makes it more obvious that it's going to iterate forever.

Reply:
I kept the mini ELMo to 32 but reduced all other dimensions. The test is pretty fast (few seconds)